consensus/istanbul/backend/engine.go

```diff
@@ -335,7 +335,9 @@ func (sb *Backend) VerifySeal(chain consensus.ChainReader, header *types.Header)
 // rules of a particular engine. The changes are executed inline.
 func (sb *Backend) Prepare(chain consensus.ChainReader, header *types.Header) error {
 	// unused fields, force to set to empty
-	header.Coinbase = sb.address
+	if header.Coinbase == (common.Address{}) {
```

Comment:
Can you explain to me why this needs to change?

Reply:
Without this in place if a proposer tries to set their own `header.Coinbase` with `txFeeRecipient` it will be overwritten here. This enforces that if the proposer has set `header.Coinbase` intentionally, do not overwrite.